### PR TITLE
graphQl-890: Replaced usage of the CartItemQuantity with the CartItem…

### PR DESCRIPTION
**Pull request by VitaliyBoyko**
…Interface
### Description (*)
Replaced usage of the CartItemQuantity with the CartItemInterface in the GraphQl scheme.

### Fixed Issues (if relevant)
1. #890: [Checkout] Replace usage of CartItemQuantity with CartItemInterface

### CODE REVIEW

**Review by lenaorobei — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @lenaorobei, thank you for the review.
[ENGCOM-5772](https://jira.corp.magento.com/browse/ENGCOM-5772) has been created to process this Pull Request
:eight_spoked_asterisk: @lenaorobei, could you please add one of the following labels to the Pull Request?

| Label | Description |
|--------------------------------|-----------------------------------------------------------------|
| ```Auto-Tests: Covered``` | All changes in Pull Request is covered by auto-tests |
| ```Auto-Tests: Not Covered``` | Changes in Pull Request requires coverage by auto-tests |
| ```Auto-Tests: Not Required``` | Changes in Pull Request does not require coverage by auto-tests |

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Could you please pull latest changes from `2.3-develop`

**Comment by VitaliyBoyko** [comment]
@lenaorobei Done

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 1]
```diff
@@ -217,16 +217,16 @@ interface CartAddressInterface @typeResolver(class: "\\Magento\\QuoteGraphQl\\Mo
 type ShippingCartAddress implements CartAddressInterface {
     available_shipping_methods: [AvailableShippingMethod] @resolver(class: "\\Magento\\QuoteGraphQl\\Model\\Resolver\\ShippingAddress\\AvailableShippingMethods")
     selected_shipping_method: SelectedShippingMethod @resolver(class: "\\Magento\\QuoteGraphQl\\Model\\Resolver\\ShippingAddress\\SelectedShippingMethod")
-    items_weight: Float
-    cart_items: [CartItemQuantity]
     customer_notes: String
+    items_weight: Float @deprecated
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: Please add `reason` here.

File: app/code/Magento/QuoteGraphQl/etc/schema.graphqls [hunk 2]
```diff
@@ -217,16 +217,16 @@ interface CartAddressInterface @typeResolver(class: "\\Magento\\QuoteGraphQl\\Mo
...
+    cart_items: [CartItemInterface]
 }
 
 type BillingCartAddress implements CartAddressInterface {
     customer_notes: String @deprecated (reason: "The field is used only in shipping address")
 }
 
-type CartItemQuantity {
+type CartItemQuantity @deprecated(reason: "Use CartItemInterface instead") {
```
Comment (lenaorobei) [inline app/code/Magento/QuoteGraphQl/etc/schema.graphqls]: From the ticket: 
> All fields in CartItemQuantity should be deprecated (so this type can be completely removed in the future releases)

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please address comments.

**Review by lenaorobei — CHANGES_REQUESTED** [review]
Please add test case to check `shipping_addresses` field.

**Review by lenaorobei — APPROVED** [review]

**Pull request merged**